src/Operations.jl

```diff
@@ -1573,17 +1573,28 @@ function free(ctx::Context, pkgs::Vector{PackageSpec}; err_if_free=true)
     end
 end
 
-function gen_test_code(source_path::String;
-        coverage=false,
-        julia_args::Cmd=``,
-        test_args::Cmd=``)
+function gen_test_code(source_path::String; coverage, julia_args::Cmd, test_args::Cmd)
     test_file = testfile(source_path)
     code = """
         $(Base.load_path_setup_code(false))
         cd($(repr(dirname(test_file))))
         append!(empty!(ARGS), $(repr(test_args.exec)))
         include($(repr(test_file)))
         """
+    return gen_subprocess_cmd(code; coverage, julia_args)
+end
+
+function gen_test_precompile_code(; coverage, julia_args::Cmd, test_args::Cmd)
+    code = """
+        $(Base.load_path_setup_code(false))
```

Comment:
I don't get why this is needed. If the tests start with the correct project, why isn't just a `Pkg.precompile` enough?

Reply:
I assumed that importing `Pkg` into the test process might be breaking, so I left that untouched